pkg/operator/csr/cert_controller.go

```diff
@@ -265,6 +278,15 @@ func (c *clientCertificateController) syncCSR(secret *corev1.Secret) (map[string
 	return data, nil
 }
 
+func (c *clientCertificateController) ensureTLSMetadata(ctx context.Context, secret *corev1.Secret, exists bool) error {
+	if certrotation.NeedsTLSMetadataUpdate(&secret.ObjectMeta, c.JiraComponent, c.Description) && exists {
+		if err := c.saveSecret(secret); err != nil {
```

Comment:
this construction will cause the the saveSecret to get a conflict error, right?  You probably want this on an `else` for the c.csrName and you probably want the current impl to take into account both the syncCSR and the annotation state.

This particular controller can probably be split out from the others if you like.